python/bufr/obs_builder/obs_builder.py

```diff
@@ -41,11 +39,20 @@ def __init__(self,
         self.log = Logger(log_name)
         self.config = config
         self.description = self._make_description()
-        self.uses_categories = uses_categories
-        self.uses_cache = uses_cache
 
     # Virtual Method
     def make_obs(self, comm, input : Union[str, dict]) -> bufr.DataContainer:
+        """
+        This method is the main method that can be overridden (optional). Its objective is to read
+        the bufr file and to create an DataContainer object and return it. The data container is
+        used in conjunction with the encoder Descriptor to encode the data into the format of your
+        choice.
+
+        :param comm: MPI communicator
+        :param input: Either a path to the input data or a dictionary the maps the input data to the
+                      mapping file paths (see constructor)
+        :return: DataContainer object
+        """
         if not isinstance(input, str) or len(self.map_dict) != 1:
             assert False, 'You must create a custom override for make_obs().'
```

Comment:
How about changing to the following?
raise NotImplementedError("Please override make_obs() when using multiple mapping files or non-string input.")

Reply:
Ok, changed it.